src/app/features/stacks-transaction-request/hooks/use-stacks-transaction-summary.ts

```diff
@@ -59,19 +58,21 @@ export function useStacksTransactionSummary(token: CryptoCurrency) {
     };
   }
 
+  // This function violates the rule of not pre-formatting data
   function formReviewTxSummary(tx: StacksTransactionWire, symbol = 'STX', decimals = 6) {
     if (symbol !== 'STX') {
       return formSip10TxSummary(tx, symbol, decimals);
     }
 
+    // WARNING: this is a very dangerous type cast as it can totally be another tx type...
```

Comment:
Should we stop to fix this now since appears to have been there prior to stacks.js upgrade?

Reply:
Yeah would be good idea.